services/order.py

```diff
@@ -0,0 +1,44 @@
+import datetime
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order, User, MovieSession
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime = None
+) -> Order:
+    with transaction.atomic():
+        user = User.objects.get(username=username)
+        if date:
+            order = Order.objects.create(user=user)
+            order.created_at = date
+            order.save()
+        else:
+            order = Order.objects.create(user=user)
+
+        for ticket_data in tickets:
+            movie_session_id = ticket_data["movie_session"]
+            row = ticket_data.get("row")
+            seat = ticket_data.get("seat")
+
+            if row is not None and seat is not None:
+                movie_session = MovieSession.objects.get(pk=movie_session_id)
+                ticket = Ticket(order=order,
+                                movie_session=movie_session,
+                                row=row, seat=seat
+                                )
+                ticket.full_clean()
+                ticket.save()
+
+    return order
+
+
+def get_orders(username: str = None) -> QuerySet:
+    if username:
+        return Order.objects.filter(user__username=username)
+    else:
```

Comment:
you don't need to specify an alternative action because you can simply return all orders. Therefore, there's no need for an explicit `else` block.